bundle/edu.gemini.itc.shared/src/main/scala/edu/gemini/itc/shared/InstrumentDetails.scala

```diff
@@ -82,3 +82,28 @@ final case class GemsParameters(
                      avgStrehl:           Double,
                      strehlBand:          String)
 
+
+object InstrumentDetails {
+
+  // NOTE: This is similar to the code in ItcUniqueConfig which decides on imaging or spectroscopy setup
+  // on Config elements. There should be a way to share this?
```

Comment:
Regarding this see also my comments in: http://jira.gemini.edu:8080/browse/OCSADV-267 . I agree that a next refactoring step with ITC could be to use the definition of a step from the SequenceModel NG as the input values.

Regarding sealed: yeah, that surprised me, too. It compiles and works. The compiler produces a warning that the `case` statements with `MichelleParameters`/`TrecsParameters` can not be reached (because they are not defined in the `InstrumentDetails` code file, but everything works as expected. I will clean that up at some point, however Michelle and TReCS have no priority because they are not in use anymore.